app/src/main/java/com/spoony/spoony/core/designsystem/component/textField/SpoonyLargeTextField.kt

```diff
@@ -0,0 +1,212 @@
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.spoony.spoony.R
+import com.spoony.spoony.core.designsystem.theme.SpoonyAndroidTheme
+
+@Composable
+fun SpoonyLargeTextField(
+    value: String,
+    onValueChanged: (String) -> Unit,
+    placeholder: String,
+    modifier: Modifier = Modifier,
+    maxLength: Int = Int.MAX_VALUE,
+    minLength: Int = 0,
+    minErrorText: String? = null,
+    maxErrorText: String? = null
+) {
+    var isFocused by remember { mutableStateOf(false) }
+    var isError: Boolean by remember { mutableStateOf(false) }
+    var selectText: Boolean by remember { mutableStateOf(false) }
+    val spoonyColors = SpoonyAndroidTheme.colors
+
+    val borderColor = remember(isError, isFocused) {
+        when {
+            isError -> spoonyColors.error400
+            isFocused -> spoonyColors.main400
+            else -> spoonyColors.gray100
+        }
+    }
+
+    val subContentColor = remember(isError) {
+        when {
+            isError -> spoonyColors.error400
+            else -> spoonyColors.gray500
+        }
+    }
+
+    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        CustomBasicTextField(
+            value = value,
+            placeholder = placeholder,
+            onValueChanged = { newText ->
+                isError = (newText.length > maxLength || newText.trim().length < minLength)
+                selectText = newText.length > maxLength
+                if (newText.length <= maxLength) {
+                    onValueChanged(newText)
+                }
+            },
+            maxLength = maxLength,
+            borderColor = borderColor,
+            subContentColor = subContentColor,
+            modifier = modifier,
+            onFocusChanged = {
+                isFocused = it
+                if (value.length >= minLength) {
+                    isError = false
+                }
+            }
+        )
+
+        if (((minErrorText != null || maxErrorText != null) && isError)) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.spacedBy(6.dp)
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_error_24),
+                    contentDescription = null,
+                    tint = subContentColor,
+                    modifier = Modifier.size(16.dp)
+                )
+                (if (selectText) maxErrorText else minErrorText)?.let {
+                    Text(
+                        text = it,
+                        style = SpoonyAndroidTheme.typography.caption1m,
+                        color = subContentColor
+                    )
+                }
```

Comment:
p3) let을 사용하게 되면 내부 로직상 nullable한지 검사를 하는 변수가 하나 추가되게 됩니다.
또한 lambda를 생성하는 과정에서 jvm에서 미미하지만 불필요한 오버헤드가 발생할 수 있어요.
그렇기 때문에 단순 nullable 검사라면 let을 사용하지 않는 것을 권장합니다.

또한 가독성이 좋지 못하므로 아래처럼 작성하는 방법을 제안해요.
물론 변수를 하나 선언하기 때문에 초기 목적이었던 성능 개선에는 영향을 크게 주지는 못하지만, lambda를 선언하는 비용을 줄일 수 있고 가독성이 높아진다는 장점이 있습니다.

```suggestion
val errorText = if (selectText) maxErrorText else minErrorText

if (errorText != null) {
    Text(
        text = errorText,
        style = SpoonyAndroidTheme.typography.caption1m,
        color = subContentColor
    )
}
```

Reply:
감사합니다, 수정했습니다!